src/XAD/Complex.hpp

```diff
@@ -1375,6 +1375,17 @@ XAD_INLINE xad::FReal<T> norm(const complex<xad::FReal<T>>& x)
     return ::xad::detail::norm_impl(x);
 }
 
+// appleclang15 needs this overload for type paramed norm
+#if defined(__APPLE__) && defined(__clang__)
+#if defined(__apple_build_version__) && (__apple_build_version__ >= 15000000)
+template <class T>
+XAD_INLINE T norm(complex<T>& x)
+{
+    return ::xad::detail::norm_impl(x);
+}
```

Comment:
I'm a bit worried here that this function will be called for regular `double` types as well. The way it's declared, it looks like it. And XAD should not change the meaning of `std::norm(std::complex<double> x)`. Only for XAD types. Might get ambiguous overload errors in this scenario:

```c++
#include <XAD/Complex.hpp>
#include <XAD/StdCompatibility.hpp>
#include <complex>

int main() {
  std::complex<double> z(1.0, -2.3);
  
  auto res = norm(z);
}
```

Can you first of all check if the above compiles with your change? And if not, add it as a unit test (failing first). Then try to fix it. 

If it compiles, please use the debugger to confirm that it actually calls the std::norm implementation in libc++, and not XAD's implementation.

Reply:
The program builds. According to GDB the compiler does use `XAD/Complex`'s norm implementation. 

I made a small tweak to my overload and the appropriate overload is now selected by the compiler in similar scenarios.